frontend/src/views/housing/project/ProjectView.vue

```diff
@@ -0,0 +1,407 @@
+  { label: 'Applications and Permits', route: RouteName.HOUSING_PROJECTS_LIST },
+  { label: getSubmission?.value?.projectName ?? '', class: 'font-bold' }
+]);
+const loading: Ref<boolean> = ref(true);
+
+const permitsNeeded = computed(() => {
+  return permitFilter({
+    permitNeeded: PermitNeeded.YES,
+    permitStatus: PermitStatus.NEW,
+    permits: getPermits.value,
+    permitTypes: getPermitTypes.value
+  })
+    .sort(permitNameSortFcn)
+    .sort(permitBusinessSortFcn);
+});
+const permitsNotNeeded = computed(() => {
+  return permitFilter({
+    permitNeeded: PermitNeeded.NO,
+    permits: getPermits.value,
+    permitTypes: getPermitTypes.value
+  })
+    .sort(permitNameSortFcn)
+    .sort(permitBusinessSortFcn);
+});
+const permitsSubmitted: ComputedRef<Array<CombinedPermit>> = computed(() => {
+  let firstFilter = permitFilter({
+    permitNeeded: PermitNeeded.YES,
+    permitStatus: PermitStatus.APPLIED,
+    permits: getPermits.value,
+    permitTypes: getPermitTypes.value
+  });
+  let secondFilter = permitFilter({
+    permitNeeded: PermitNeeded.YES,
+    permitStatus: PermitStatus.COMPLETED,
+    permits: getPermits.value,
+    permitTypes: getPermitTypes.value
+  });
+  return firstFilter.concat(secondFilter).sort(permitNameSortFcn).sort(permitBusinessSortFcn);
+});
+
+// Actions
+function permitBusinessSortFcn(a: CombinedPermit, b: CombinedPermit) {
+  return a.businessDomain > b.businessDomain ? 1 : -1;
+}
+function permitNameSortFcn(a: CombinedPermit, b: CombinedPermit) {
+  return a.name > b.name ? 1 : -1;
+}
+function permitFilter(config: PermitFilterConfig) {
+  const { permitNeeded, permits, permitStatus, permitTypes } = config;
+  let returnArray: Array<any> = permits;
+
+  if (permitNeeded) {
+    returnArray = returnArray.map((p) => {
+      const pType = permitTypes.find((pt) => pt.permitTypeId === p?.permitTypeId && p.needed === permitNeeded);
+      if (pType) return { ...p, ...pType };
+    });
+  }
+
+  if (permitStatus) {
+    returnArray = returnArray.map((p) => {
+      const pType = permitTypes.find((pt) => pt.permitTypeId === p?.permitTypeId && p.status === permitStatus);
+      if (pType) return { ...p, ...pType };
+    });
+  }
+
+  return returnArray.filter((pt) => !!pt) as Array<CombinedPermit>;
+}
+
+function displayTrackerStatus(authStatus: string | undefined, permitState: string | undefined): string {
+  if (permitState === PermitStatus.APPLIED) {
+    switch (authStatus) {
+      case PermitAuthorizationStatus.IN_REVIEW:
+      case PermitAuthorizationStatus.PENDING:
+        return authStatus;
+      case PermitAuthorizationStatus.NONE:
+      default:
+        return 'Submitted';
+    }
+  }
+  if (permitState === PermitStatus.COMPLETED) {
+    switch (authStatus) {
+      case PermitAuthorizationStatus.ISSUED:
+      case PermitAuthorizationStatus.DENIED:
+        return authStatus;
+      default:
+        return PermitStatus.COMPLETED;
+    }
+  }
+  return 'Submitted';
+}
```

Comment:
Optional Style: This function might be something that could be redefined as a constant attribute to value mapping set to help reduce cognitive load interpreting the conditional switch cases. It'll generally be easier to comprehend at a glance explicit enumerations of `permitState` x `authStatus` and their => result in an array (and down the line unit testable) than having it be determined by switch cases and some cascading behavior.

Reply:
I think my original approach did use a attribute to value mapping. However, I can no longer recall why that approach was not suitable. It was over three weeks ago and I am hesitant to change working and tested code. Perhaps for a future update.